std/algorithm/iteration.d

```diff
@@ -1524,14 +1540,26 @@ private struct GroupByImpl(alias pred, Range)
     }
 
     @property bool empty() { return impl.current.empty; }
-    @property auto front() { return Group(impl); }
+
+    @property auto front()
+    {
+        static if (isUnary)
+        {
+            import std.typecons : tuple;
+            return tuple(unaryFun!pred(impl.current.front), Group(impl));
```

Comment:
Yep, that's my thinking. It's an optimization, and Andrei wanted this API change ASAP for 2.067. I'll see if I can make the optimization, but this is (hopefully) viable for pull.